main/guides/getting-started/swaparoo-how-to-swap-assets-explainer.md

```diff
@@ -1,5 +1,106 @@
 # Swaparoo Contract
 
-## How to Swap Assets
+This smart contract is designed to allow two parties to swap assets between themselves, with a fee charged to one of the parties. The contract is started with a `feeAmount` and a `namesByAddressAdmin` object, which is used to retrieve a deposit facet for the second party.
+
+Let's take a look at how this contract works:
+
+## Setting up Fee Handling
+The contract retrieves the `feeIssuer` from the Zoe service, which is an issuer for the stable coin used for fees. It creates a `feeSeat` and a `feeShape` based on the `feeAmount` specified in the contract terms.
+```js
+const stableIssuer = await E(zcf.getZoeService()).getFeeIssuer();
+const feeBrand = await E(stableIssuer).getBrand();
+const { zcfSeat: feeSeat } = zcf.makeEmptySeatKit();
+const feeShape = makeNatAmountShape(feeBrand, feeAmount.value);
+```
+
+## Making the First Invitation
+The `makeFirstInvitation` function is called with an array of issuers. It verifies that these issuers are part of the contract terms and saves any new issuers to the contract. It then creates an invitation with a proposal shape that includes the `feeShape` in the give record.
+```js
+const makeFirstInvitation = issuers => {
```

Comment:
_just a thought..._

I wonder how to keep these code excerpts in sync as dapp-agoric-basics changes.

The whole zoe section is wildly out of date due to using game terminology when dapp-offer-up has moved on :-/

Reply:
Would a Github workflow in the app repos that create an issue in the documentation repo when something changes work?